src/overlays/actors/ovl_En_Mu/z_en_mu.c

```diff
@@ -44,27 +34,182 @@ static ColliderCylinderInit D_80AB0BD0 = {
+CollisionCheckInfoInit2 D_80AB0BFC = { 0, 0, 0, 0, MASS_IMMOVABLE };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0724.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/EnMu_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB08A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0994.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB09A8.s")
+const ActorInit En_Mu_InitVars = {
+    ACTOR_EN_MU,
+    ACTORCAT_NPC,
+    FLAGS,
+    OBJECT_MU,
+    sizeof(EnMu),
+    (ActorFunc)EnMu_Init,
+    (ActorFunc)EnMu_Destroy,
+    (ActorFunc)EnMu_Update,
+    (ActorFunc)EnMu_Draw,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/EnMu_Draw.s")
+void EnMu_SetupAction(EnMu* this, EnMuActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnMu_Interact(EnMu* this, GlobalContext* globalCtx) {
+    u8 textIdOffset[] = { 0x42, 0x43, 0x3F, 0x41, 0x3E };
+    u8 bitmask[] = { 0x01, 0x02, 0x04, 0x08, 0x10 };
+    u8 textFlags;
+    s32 randomIndex;
+    s32 i;
+
+    textFlags = gSaveContext.eventInf[2] & 0x1F;
+    gSaveContext.eventInf[2] &= ~0x1F;
+    randomIndex = (globalCtx->state.frames + (s32)(Rand_ZeroOne() * 5.0f)) % 5;
+
+    for (i = 0; i < 5; i++) {
+
+        if (!(textFlags & bitmask[randomIndex])) {
+            break;
+        }
+
+        randomIndex++;
+        if (randomIndex >= 5) {
+            randomIndex = 0;
+        }
+    }
+
+    if (i == 5) {
+        if (this->defFaceReaction == (textIdOffset[randomIndex] | 0x7000)) {
+            if (++randomIndex >= 5) {
+                randomIndex = 0;
+            }
+        }
+        textFlags = 0;
+    }
+
+    textFlags |= bitmask[randomIndex];
+    this->defFaceReaction = textIdOffset[randomIndex] | 0x7000;
+    textFlags &= 0xFF;
+    gSaveContext.eventInf[2] |= textFlags;
+}
+
+u16 EnMu_GetFaceReaction(GlobalContext* globalCtx, Actor* thisx) {
+    EnMu* this = THIS;
+    u16 faceReaction = Text_GetFaceReaction(globalCtx, this->actor.params + 0x3A);
+
+    if (faceReaction != 0) {
+        return faceReaction;
+    }
+    return this->defFaceReaction;
+}
+
+s32 EnMu_CheckDialogState(GlobalContext* globalCtx, EnMu* this) {
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 3:
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+        case 9:
+            return 1;
+        case 2:
+            EnMu_Interact(this, globalCtx);
+            return 0;
+        default:
+            return 1;
+    }
+}
+
+void EnMu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnMu* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 160.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06004F70, &D_060003F4, 0, 0, 0);
```

Comment:
the third and second to last args are pointers so they should be NULL instead of 0